frontend/packages/dev-console/src/components/topology/topology-types.ts

```diff
@@ -66,16 +66,16 @@ export interface TopologyDataModel {
 }
 
 export type TopologyOverviewItem = OverviewItem & {
-  pipelines: Pipeline[];
-  pipelineRuns: PipelineRun[];
+  pipelines?: Pipeline[];
+  pipelineRuns?: PipelineRun[];
```

Comment:
Optional? I think they are always there, just empty arrays.

Reply:
kept it optional as wanted to use `createTopologyNodeData ` and i thought these values will only be there if pipelines operators are installed.

https://github.com/openshift/console/pull/3111/files#diff-c8b822bac394ac5898fa58d8b3cd609bR89-R90

and first argument is of type `TopologyOverviewItem` so made it optional as here we are passing default values to [] for pipelines and pipelineRuns. 

https://github.com/openshift/console/pull/3111/files#diff-6710cf4682ac7597af9d6f5e58a10860R284

Reply:
as I see here https://github.com/openshift/console/blob/master/frontend/packages/dev-console/src/plugin.tsx#L175-L179

```
{
    type: 'Overview/CRD',
    properties: {
      resources: tknPipelineAndPipelineRunsResources,
      required: SHOW_PIPELINE,
      utils: getPipelinesAndPipelineRunsForResource,
    }
```

it's based on feature flag `SHOW_PIPELINE` , so having it as optional i thought would make sense, WDYT

Reply:
Yup, that's probably fair. Carry on :) I retract my objection.

Reply:
They were given default values as well:
```
    pipelines = [],
    pipelineRuns = [],
```
So it makes sense they are optional. Otherwise the defaults will never be a thing.